cmd/juju/status/status_internal_test.go

```diff
@@ -6077,6 +6082,32 @@ func (s *StatusSuite) TestNonTabularRelations(c *gc.C) {
 	c.Assert(strings.Contains(string(stdout), "storage:"), jc.IsTrue)
 }
 
+func (s *StatusSuite) TestBranchesOutput(c *gc.C) {
+	ctx := s.FilteringTestSetup(c)
+	addBranch{"test"}.step(c, ctx)
+	addBranch{"bla"}.step(c, ctx)
+	ct, err := s.ControllerStore.CurrentController()
+	c.Assert(err, jc.ErrorIsNil)
+	md, err := s.ControllerStore.CurrentModel(ct)
+	c.Assert(err, jc.ErrorIsNil)
+	m, err := s.JujuConnSuite.ControllerStore.ModelByName(ct, md)
+	c.Assert(err, jc.ErrorIsNil)
+	m.ActiveBranch = "bla"
+	err = s.ControllerStore.UpdateModel(ct, md, *m)
+	c.Assert(err, jc.ErrorIsNil)
+
+	defer s.resetContext(c, ctx)
+
+	_, stdout, stderr := runStatus(c, "--format=yaml")
+	c.Assert(stderr, gc.IsNil)
+	c.Assert(strings.Contains(string(stdout), "active: true"), jc.IsTrue)
+
+	_, stdout, stderr = runStatus(c)
```

Comment:
Typically we try to only test 1 thing per unit test.  So this would be broken up into TestStatusFormatTabularBranches and TestStatusFormatYamlBranches.  The setup could be broken out into a separate method for both to use.

Reply:
:+1: